rust/types/src/collection_configuration.rs

```diff
@@ -207,19 +197,38 @@ impl InternalCollectionConfiguration {
     pub fn try_from_config(
         value: CollectionConfiguration,
         default_knn_index: KnnIndex,
+        metadata: Option<Metadata>,
     ) -> Result<Self, CollectionConfigurationToInternalConfigurationError> {
-        match (value.hnsw, value.spann) {
+        let mut hnsw: Option<HnswConfiguration> = value.hnsw;
+        let spann: Option<SpannConfiguration> = value.spann;
+
+        if hnsw.is_none() && spann.is_none() {
+            let hnsw_config_from_metadata =
```

Comment:
after this we should cleanup the sysdb. To confirm, we can simple remove that part and assume that upstream callers to sysdb always set the config?

Reply:
Also, please add some comments here